state/accountsDBApi.go

```diff
@@ -154,6 +154,11 @@ func (accountsDB *accountsDBApi) GetCode(codeHash []byte) []byte {
 
 // RootHash will return last loaded root hash
 func (accountsDB *accountsDBApi) RootHash() ([]byte, error) {
+	_, err := accountsDB.recreateTrieIfNecessary()
```

Comment:
Incorrect change (dangerous approach), might affect other endpoints of the Node API, since `RootHash()` is a commonly used. Please adjust to be consistent with other API endpoints.